### PR TITLE
add checks for undefined transitions

### PR DESCRIPTION
**Pull request by boygirl**
cc/ @divmain 

I was getting some failures when transitions were not defined. This PR adds some checks and fallback behaviors.

### CODE REVIEW

File: src/victory-util/transitions.js [hunk 1]
```diff
@@ -107,27 +107,27 @@ export function getInitialTransitionState(oldChildren, nextChildren) {
 
 
 function getInitialChildProps(animate, data) {
+  const before = animate.onExit && animate.onExit.before ? animate.onExit.before : identity;
```
Comment (divmain) [inline src/victory-util/transitions.js:110]: Nice!

**Comment by divmain** [comment]
Yep, this looks good to me!

**Comment by boygirl** [comment]
I can't figure out why CI is failing in this branch, but it seems quite unrelated to the changes

**Comment by boygirl** [comment]
I'm going to merge / release this change anyway, since it's blocking other work. I'm seeing this CI failure in other repos too... Will investigate.

**Pull request merged**